pkg/stub/handler.go

```diff
@@ -2,96 +2,436 @@ package stub
+			if err != nil {
+				return h.processError(opcfg, v1alpha1.SamplesUpdateFailed, err, "error determining distro/type : %v")
+			}
+			files, err := h.filefinder.List(dir)
+			if err != nil {
+				return h.processError(opcfg, v1alpha1.SamplesUpdateFailed, err, "error reading in content : %v")
+			}
+			err = h.processFiles(dir, files, opcfg)
+			if err != nil {
+				return h.processError(opcfg, v1alpha1.SamplesUpdateFailed, err, "error processing content : %v")
+			}
+
 		}
-		logrus.Printf("template client %#v", tempclient)
-		imageclient, err := getImageClient(restconfig)
+
+		now := kapis.Now()
+		status := opcfg.Condition(v1alpha1.SamplesExists)
+		status.LastUpdateTime = now
+		if status.Status != newStatus {
+			status.Status = newStatus
+			status.LastTransitionTime = now
+		}
+		opcfg.ConditionUpdate(status)
+		err := h.sdkwrapper.Update(opcfg)
 		if err != nil {
-			logrus.Errorf("failed to get image client : %v", err)
-			return err
-		}
-		logrus.Printf("image client %#v", imageclient)
-		templist, err := tempclient.Templates("openshift").List(metav1.ListOptions{})
-		logrus.Printf("template list %#v", templist)
-		/*err := sdk.Create(newbusyBoxPod(o))
-		if err != nil && !errors.IsAlreadyExists(err) {
-			logrus.Errorf("failed to create busybox pod : %v", err)
-			return err
-		}*/
+			if !kerrors.IsConflict(err) {
+				return h.processError(opcfg, v1alpha1.SamplesUpdateFailed, err, "failed adding success condition to config: %v")
+			}
+			logrus.Println("got conflict error on success status update, just going to forgo this time vs. retry")
+		}
+
+		h.operatorCfg = opcfg
+
+		logrus.Println("")
+		logrus.Println("")
+		logrus.Println("")
+		logrus.Println("")
+	}
+	return nil
+}
+
+func (h *Handler) buildSkipFilters(opcfg *v1alpha1.SamplesResource) {
+	for _, st := range opcfg.Spec.SkippedTemplates {
+		h.skippedTemplates[st] = true
+	}
+	for _, si := range opcfg.Spec.SkippedImagestreams {
+		h.skippedImagestreams[si] = true
+	}
+}
+
+func (h *Handler) processError(opcfg *v1alpha1.SamplesResource, ctype v1alpha1.SamplesResourceConditionType, err error, msg string, args ...interface{}) error {
+	log := ""
+	if args == nil {
+		log = fmt.Sprintf(msg, err)
+	} else {
+		log = fmt.Sprintf(msg, err, args)
+	}
+	logrus.Println(log)
+	var status *v1alpha1.SamplesResourceCondition
+	switch ctype {
+	case v1alpha1.SecretUpdateFailed:
+		status = opcfg.Condition(v1alpha1.ImportCredentialsExists)
+	case v1alpha1.SamplesUpdateFailed:
+		status = opcfg.Condition(v1alpha1.SamplesExists)
+	default:
+		return err
+	}
+	now := kapis.Now()
+	status.LastUpdateTime = now
+	if status.Status != corev1.ConditionUnknown {
+		status.Status = corev1.ConditionUnknown
+		status.LastTransitionTime = now
+	}
+	status.Message = log
+	opcfg.ConditionUpdate(status)
+	err2 := h.sdkwrapper.Update(opcfg)
+	if err2 != nil {
+		// just log this error
+		logrus.Printf("failed to add error condition to config status: %v", err2)
+	}
```

Comment:
same here, wrap this inside the if status!=Unknown check.  No reason to update it if it isn't changing.

Reply:
(Unless you're going to start setting the status.LastUpdateTime every time you re-assert the same status, but i'm concerned about us spamming the api server if we start doing that)

Reply:
OK that answers my earlier question ... will not worry about last update time.